include/glow/Graph/Graph.h

```diff
@@ -140,6 +140,15 @@ class Module final {
                  float val = 0.0);
   ///@}
 
+  /// Variable builder for a singleton Variable with a reserved \p name,
+  /// reserved in the Module constructor. A singleton Variable with this name
+  /// should only be created once.
+  Variable *createSingletonVariable(
+      ElemKind T, llvm::ArrayRef<size_t> dims, float scale, int32_t offset,
+      llvm::StringRef name, VisibilityKind visibility = VisibilityKind::Private,
+      Variable::TrainKind train = Variable::TrainKind::Broadcast,
+      float val = 0.0);
```

Comment:
What happens if you try to create a singleton variable twice?
It's easy to see in the code, but we shouldn't have to check the code to have the answer :).

Reply:
Absolutely can update the comment!

But, I was under the impression @nadavrot preferred to only do https://github.com/pytorch/glow/pull/1263 to deduplicate these Variables, and wanted to abandon this PR.

Of course, this PR would deduplicate these Vars at creation time and avoid some optimization time, so I do think it has value aside from https://github.com/pytorch/glow/pull/1263.